### DISCUSSION TITLE
Is it possible to run passage retrieval models?

### DISCUSSION THREAD

**Discussion by semoal** [michaelfeil/infinity#120] [Q&A]
We would like to run this model to do hybrid search, https://huggingface.co/naver/efficient-splade-VI-BT-large-doc, is it possible with the current API or should we introduce another method like `classify`, `rerank`, add another for `mask`?

**Comment by semoal** [comment]
![image](https://github.com/michaelfeil/infinity/assets/22656541/32307b30-c6e2-40a0-b58b-56d4818bb3e0)

Tried calling `embed` and looks quite good;  gonna test it with our current RAG to see if its enough

**Reply by michaelfeil** [reply]
Have you tried to assert the difference between the original implementation and infinity? Can you load it via sentence transformers? Maybe you need to choose the mean/cls pooling method, or add a sentence transformers config in the upstream repo.

i think infinity itself is compatible, it might need adaption to be consumable from sentence transformers.

**Comment by ouz-m** [comment]
@semoal hi, do you have any updates on this topic? I am curious to hear how it turned out!
